lib/index.js

```diff
@@ -262,6 +262,23 @@ class Study extends EventTarget {
         die();
       }).bind(this)
     )
+    this.on(
+      "uninstall-custom-reason", (function (reason) {
+        prefSvc.get('shield.debug') && console.log("uninstall-custom-reason", this.states);
+        this.states.push("uninstall-custom-reason");
+        if (this.flags.dying) {
+          this.final();
+          return;
+        }
+        this.flags.dying = true;
+        this.report(merge({}, telemetrySubset(this.config), {study_state: reason}), "shield");
```

Comment:
When I spot-checked calling `shutdown()` with a name matching a key of the `surveyUrls`, I got:
```
console.error: shield-study-privacy:
JPM [error]   Message: Error: The state cannot be set or get. The object may be not be registered, or may already have been unloaded.
```
from this line. Does `this.states.push()` and/or `this.final()` maybe destroy the `TelemetryController`? Can you show me some code that spot-checks this outside of the test code?